algorithms.py

```diff
@@ -1553,10 +1610,14 @@ def merge_ancestors(self, H, pop_id, label):
             if len(X) == 1:
                 x = X[0]
                 if len(H) > 0 and H[0][0] < x.right:
+                    next_l_mass = self.recomb_map.position_to_mass(H[0][0])
```

Comment:
OK, so this is the only lookup we need to do. But, aren't all the values in H a result of being either left or right coordinates? Whenever we push a coordinate into H there's also a segment along with it, so we can get the mass from there, avoiding this lookup.